plugins/workspace-backpack/src/backpack.js

```diff
@@ -39,114 +39,108 @@ export class Backpack {
+     * The backpack flyout. Initialized during init.
+     * @type {?Blockly.IFlyout}
      * @protected
      */
-    this.svgGroup_ = null;
-
-    /**
-     * Left coordinate of the backpack.
-     * @type {number}
-     * @private
-     */
-    this.left_ = 0;
-
-    /**
-     * Top coordinate of the backpack.
-     * @type {number}
-     * @private
-     */
-    this.top_ = 0;
+    this.flyout_ = null;
 
     /**
-     * Extent of hotspot on all sides beyond the size of the image.
-     * @const {number}
-     * @private
+     * A list of XML (stored as strings) representing blocks in the backpack.
+     * @type {!Array<string>}
+     * @protected
      */
-    this.HOTSPOT_MARGIN_ = 10;
+    this.contents_ = [];
 
     /**
-     * Top offset for backpack in svg.
-     * @type {number}
-     * @private
+     * Array holding info needed to unbind events.
+     * Used for disposing.
+     * @type {!Array<!Blockly.browserEvents.Data>}
+     * @protected
      */
-    this.SPRITE_TOP_ = 10;
+    this.boundEvents_ = [];
 
     /**
-     * Left offset for backpack in svg.
+     * Left coordinate of the backpack.
      * @type {number}
-     * @private
+     * @protected
      */
-    this.SPRITE_LEFT_ = 20;
+    this.left_ = 0;
 
     /**
-     * Width/Height of svg.
+     * Top coordinate of the backpack.
      * @type {number}
-     * @const
-     * @private
+     * @protected
      */
-    this.SPRITE_SIZE_ = 80;
+    this.top_ = 0;
 
     /**
      * Width of the backpack. Used for clip path.
      * @type {number}
      * @const
-     * @private
+     * @protected
      */
     this.WIDTH_ = 40;
 
     /**
      * Height of the backpack. Used for clip path.
      * @type {number}
      * @const
-     * @private
+     * @protected
      */
     this.HEIGHT_ = 60;
 
     /**
      * Distance between backpack and bottom or top edge of workspace.
      * @type {number}
      * @const
-     * @private
+     * @protected
      */
     this.MARGIN_VERTICAL_ = 20;
 
     /**
      * Distance between backpack and right or left edge of workspace.
      * @type {number}
      * @const
-     * @private
+     * @protected
      */
     this.MARGIN_HORIZONTAL_ = 20;
+
     /**
-     * Array holding info needed to unbind events.
-     * Used for disposing.
-     * @type {!Array<!Blockly.browserEvents.Data>}
+     * Extent of hotspot on all sides beyond the size of the image.
+     * @const {number}
      * @protected
      */
-    this.boundEvents_ = [];
+    this.HOTSPOT_MARGIN_ = 10;
 
     /**
-     * Whether this has been initialized.
-     * @type {boolean}
+     * The SVG group containing the backpack.
+     * @type {?SVGElement}
      * @protected
      */
-    this.initialized_ = false;
+    this.svgGroup_ = null;
 
     /**
-     * A list of XML (stored as strings) representing blocks in the backpack.
-     * @type {!Array<string>}
-     * @protected
+     * Top offset for backpack in svg.
+     * @type {number}
+     * @private
```

Comment:
Should these be protected as well? `getTargetArea` uses them, so if someone overrides `createDom_` to use a different image, they'd either need to use override these, or else the entire `getTargetArea` function, just to change these values.

I could be missing something though

otherwise, lgtm

Reply:
The constants starting with `SPRITE_*` are specific to the Backpack svg currently used to handle the whitespace that exists around the svg.
There isn't currently a way to change the button asset without subclassing. They would need to override both `createDom_` and `getTargetArea`. I'll make a separate PR addressing adding hints to the API to help let developers know that they should be overriding both.